examples/pod_resize.rs

```diff
@@ -0,0 +1,131 @@
+    let client = Client::try_default().await?;
+
+    let pods: Api<Pod> = Api::default_namespaced(client);
+
+    // Create a sample pod with resource limits
+    info!("Creating pod with initial resource requirements");
+    let pod: Pod = serde_json::from_value(serde_json::json!({
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": { "name": "resize-demo" },
+        "spec": {
+            "containers": [{
+                "name": "app",
+                "image": "nginx:1.14.2",
+                "resources": {
+                    "requests": {
+                        "cpu": "100m",
+                        "memory": "128Mi"
+                    },
+                    "limits": {
+                        "cpu": "200m",
+                        "memory": "256Mi"
+                    }
+                }
+            }]
+        }
+    }))?;
+
+    let pp = PostParams::default();
+    pods.create(&pp, &pod).await?;
+
+    // Wait for pod to be running
+    info!("Waiting for pod to be running...");
+    let running = await_condition(pods.clone(), "resize-demo", is_pod_running());
+    let _ = tokio::time::timeout(std::time::Duration::from_secs(30), running).await?;
+
+    // Example 1: Using get_resize to view current state
+    info!("Example 1: Getting pod resize subresource");
+    let current = pods.get_resize("resize-demo").await?;
+    if let Some(spec) = &current.spec {
+        if let Some(container) = spec.containers.first() {
+            info!("Current resources: {:?}", container.resources);
+        }
+    }
+
+    // Example 2: Using patch_resize to update resources
+    info!("Example 2: Patching pod resources using resize subresource");
+    let patch = serde_json::json!({
+        "spec": {
+            "containers": [{
+                "name": "app",
+                "resources": {
+                    "requests": {
+                        "cpu": "150m",
+                        "memory": "256Mi"
+                    },
+                    "limits": {
+                        "cpu": "300m",
+                        "memory": "512Mi"
+                    }
+                }
+            }]
+        }
+    });
+
+    let patch_params = PatchParams::default();
+    match pods
+        .patch_resize("resize-demo", &patch_params, &Patch::Strategic(patch))
+        .await
+    {
+        Ok(resized) => {
+            info!("Successfully patched pod: {}", resized.name_any());
+            if let Some(spec) = resized.spec {
+                if let Some(container) = spec.containers.first() {
+                    info!("Updated resources via patch: {:?}", container.resources);
+                }
+            }
+        }
+        Err(e) => {
+            error!("Failed to patch resize pod: {}", e);
+        }
+    }
+
+    // Example 3: Using replace_resize
+    info!("Example 3: Using replace_resize method");
+    let mut current_pod = pods.get_resize("resize-demo").await?;
```

Comment:
do you think it's possible and/or worth extending the `conditions` module to be able to wait for the `PodResizePending` or / `PodResizeInProgress` [condition to complete](https://kubernetes.io/docs/tasks/configure-pod-container/resize-container-resources/#pod-resize-status)?

i feel like if you are patching things like this you might want to "wait for it to complete", and if an `await_condition` hasn't succeeded in like 10s then maybe people would look into whether the request was "Infeasible" / "Deferred" themselves.